pkg/controller/dataimportcron-controller.go

```diff
@@ -263,28 +263,28 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 
 	desiredDigest := dataImportCron.Annotations[AnnSourceDesiredDigest]
 	digestUpdated := desiredDigest != "" && (len(imports) == 0 || desiredDigest != imports[0].Digest)
+	incrementMetric := true
+	currentUpToDateCondition := FindDataImportCronConditionByType(dataImportCron, cdiv1.DataImportCronUpToDate).DeepCopy()
```

Comment:
again, why use and DeepCopy and pass the Condition if you can use a bool?
```
wasUpToDate := false 
if condition := FindDataImportCronConditionByType(cron, conditionType); condition != nil {
   wasUpToDate = condition.Status == corev1.ConditionFalse
}
```

Reply:
Tried to fiddle around with it, I prefer to keep this chunk hidden inside `updateDataImportCronCondition()` if that's ok